### PR TITLE
Try reducing concurrent downloads

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
By default the app sets the resource manager concurrent downloads to 16 (can be overridden on the command line). A quick test confirms typical domains can easily reach that limit (with several more downloads pending). This experimental PR tries reducing that to 4 for Android.

https://highfidelity.manuscript.com/f/cases/14394/Domain-disconnects-can-occur-on-low-end-machines

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9822/HighFidelity-Beta-PR13176-2b30c1389e79b9b82ba34b0af453e3be370a42ad.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9822/HighFidelity-Beta-PR13176-2b30c1389e79b9b82ba34b0af453e3be370a42ad.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9822/HighFidelity-Beta-Interface-PR13176-2b30c1389e79b9b82ba34b0af453e3be370a42ad.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9826/HighFidelity-Beta-PR13176-f4d24a3d0e9a19cb87fc760677fb49ef80b2b56a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9826/HighFidelity-Beta-PR13176-f4d24a3d0e9a19cb87fc760677fb49ef80b2b56a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9826/HighFidelity-Beta-Interface-PR13176-f4d24a3d0e9a19cb87fc760677fb49ef80b2b56a.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9835/HighFidelity-Beta-PR13176-b43cb5b86f86760bba2c0733d87e4647c2817950.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9835/HighFidelity-Beta-PR13176-b43cb5b86f86760bba2c0733d87e4647c2817950.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9835/HighFidelity-Beta-Interface-PR13176-b43cb5b86f86760bba2c0733d87e4647c2817950.dmg

**Comment by SimonWalton-HiFi** [comment]
## Test Plan

On Android with a fresh install (or cleared cache) visiting a domain at start-up should be smooth with no obvious disconnects. There should be no regressions. Ideally we should have quantitative comparisons before/after the PR.

**Comment by SamGondelman** [comment]
Why are the downloads blocking the heartbeat?

**Comment by SimonWalton-HiFi** [comment]
I think it's just network congestion and dropped packets.

**Comment by SamGondelman** [comment]
But shouldn't we be sending the heartbeat from a different thread?

**Comment by SimonWalton-HiFi** [comment]
The domain check-ins are sent from a separate thread but the network layer of the kernel is common to all processes.

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -332,7 +332,11 @@ static const QString RENDER_FORWARD{ "HIFI_RENDER_FORWARD" };
 static bool DISABLE_DEFERRED = QProcessEnvironment::systemEnvironment().contains(RENDER_FORWARD);
 #endif
 
+#if !defined(Q_OS_ANDROID)
 static const int MAX_CONCURRENT_RESOURCE_DOWNLOADS = 16;
+#else
+static const int MAX_CONCURRENT_RESOURCE_DOWNLOADS_LOWEND = 4;
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: I don't understand why this define is part of a platform `#ifdef` ?? I could understand having multiple defines for normal and lowed... and then having some other `#ifdefs` that pick if the platform is considered low end or not...

Comment (SimonWalton-HiFi) [reply interface/src/Application.cpp]: I originally did it that way but macOS gave a warning about the unused variable. I can change it to just have the one const int if you prefer.

Comment (huffman) [reply interface/src/Application.cpp]: I might not be following, but can we just name this `MAX_CONCURRENT_RESOURCE_DOWNLOADS` and not have the `ifdef` down below?

Comment (SimonWalton-HiFi) [reply interface/src/Application.cpp]: OK, I'll change it that way. It was this way just for a macOS warning.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10005/HighFidelity-Beta-PR13176-f702374db3a7e5d899e4e6468a6e1f50edb1440f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10005/HighFidelity-Beta-Interface-PR13176-f702374db3a7e5d899e4e6468a6e1f50edb1440f.dmg

**Comment by qligmaloney** [comment]
Currently testing

**Comment by qligmaloney** [comment]
PASS @conklin94122 
Loading into Dev-Mobile clone (qa_pr_metastable_android) is smooth and faster than Master build
Master:
Galaxy S8+: 1m 10s
Pixel 2: 1m 7s

PR13176:
Galaxy S8+: 31.3s
Pixel 2: 27.4s

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10198/HighFidelity-Beta-PR13176-f702374db3a7e5d899e4e6468a6e1f50edb1440f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10198/HighFidelity-Beta-PR13176-f702374db3a7e5d899e4e6468a6e1f50edb1440f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10198/HighFidelity-Beta-Interface-PR13176-f702374db3a7e5d899e4e6468a6e1f50edb1440f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10198/HighFidelity-Beta-PR13176-f702374db3a7e5d899e4e6468a6e1f50edb1440f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10198/HighFidelity-Beta-Interface-PR13176-f702374db3a7e5d899e4e6468a6e1f50edb1440f.exe

**Comment by divyaelango88** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10257/HighFidelity-Beta-PR13176-f702374.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10257/HighFidelity-Beta-PR13176-f702374.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10257/HighFidelity-Beta-Interface-PR13176-f702374.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10257/HighFidelity-Beta-PR13176-f702374.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10257/HighFidelity-Beta-Interface-PR13176-f702374.exe

**Pull request merged**